### ISSUE TITLE
Vagrant: problems with fresh setup

### ISSUE THREAD

**Issue by tmtmtmtm** [mysociety/pombola#1890]
After running a fresh `vagrant up` this morning, I was unable to get the server to start:

After ssh-ing into the box, `cd /vagrant/pombola`, followed by `python manage.py runserver 0.0.0.0:8000` errors with

```
Traceback (most recent call last):
  File "manage.py", line 4, in <module>
    import yaml
ImportError: No module named yaml
```

I got a few warnings/errors from the initial installation, though I can't see any directly related to YAML. 

The full output is at https://gist.github.com/tmtmtmtm/e6f4e93a8e60e03437e7 and some others that might imply other problems (or are likely to cause concern to anyone trying to get up and running) are:

```
==> default: bin/prepare_environment.bash: line 56: /vagrant/pombola-virtualenv/local/lib/python2.7/site-packages/mapit/../bin/make_css: No such file or directory
```

```
==> default: Missing objects with COUNTRY_APP 'kenya', skipping migration 0038
```

and 

```
==> default: ERROR:  Error installing json:
==> default: 	ERROR: Failed to build gem native extension.
==> default: 
==> default:         /opt/vagrant_ruby/bin/ruby extconf.rb
==> default: creating Makefile
==> default: 
==> default: make
==> default: sh: 1: make: not found
==> default: 
==> default: 
==> default: Gem files will remain installed in /vagrant/gems/gems/json-1.8.3 for inspection.
==> default: Results logged to /vagrant/gems/gems/json-1.8.3/ext/json/ext/generator/gem_make.out
```

**Comment by mhl** [comment]
The `No module named yaml` issue is just that the virtualenv hasn't been activated, I think - the script should point out that you have to do:

```
cd /vagrant/
. pombola-virtualenv/bin/activate
cd pombola
./manage.py runserver 0.0.0.0:8000
```

The `make_css: No such file or directory` error is more or less harmless in fact, it just means that the line in `bin/prepare_environment.bash` that's meant to build the MapIt CSS has failed (we should fix that, but it's not a big deal).

The `Missing objects with COUNTRY_APP 'kenya', skipping migration 0038` error is expected if you're running migrations from scratch rather than with existing data.

The `sh: 1: make: not found` is a problem, though - it seems that your precise64 base box doesn't have GNU `make` installed (whereas the one I'm using does) so I've pushed a change to `master` (0b516d58226757b5) to make sure it's installed.

I've also added a fix that I needed for the initial migrations to run (5461a8b000e2)

**Comment by mhl** [comment]
I've also pushed a fix for the MapIt CSS error: ebc0a3aa68933bfd98

**Comment by tmtmtmtm** [comment]
> is just that the virtualenv hasn't been activated

Yes, it looks like that's the case. I'm not sure whether "the script should point out" means that it was expected that it should, and for some reason it didn't (or that I'm not seeing it), or that it should be changed so that it does, but the current instructions are definitely confusing that regard. 

Everything ran fine this time, though there was a new error right at the end:
```
==> default: /vagrant/pombola/conf/general.yml was not a symlink, and should be
```

Also, when I ran `./manage.py runserver 0.0.0.0:8000` it took quite a long time for anything to happen, and as I didn't know what was _expected_ to happen (if anything), I wasn't sure whether I was meant to just keep waiting, or whether something had gone wrong. If there was a way to say something straightaway, rather than there being nothing until the "Validating models..." output, that could be quite helpful.

However, everything does seem to be working now (other than the fact that with no data pretty much nothing within the site actually works…)

**Comment by mhl** [comment]
>  or that it should be changed so that it does

That was the one I meant :)

I suspect the delay in starting up is largely due to a combination of (a) everything (all code and data) being shared with the host and (b) it's just using whatever Vagrant's default method for doing that sharing is, rather than NFS which is a lot faster.

> ==> default: /vagrant/pombola/conf/general.yml was not a symlink, and should be

Oops, I think I just introduced that with 5461a8b000e27a - I forgot that `general.yml` would be a symlink, and `sed -i EXPR FILE` will replace the symlink with a regular file. I'll leave this open as a reminder to fix that.

**Comment by mhl** [comment]
OK, I've fixed that last error with 2a16c1d0373277696e2

Incidentally, the final step of the installation process is to modify `~/.bashrc` to activate the virtualenv on login, so I don't think the "documentation" needs to be updated after all - so long as the installation completes correctly (as it should now) that won't be a problem.
